packages/block-library/src/navigation/index.php

```diff
@@ -250,29 +250,24 @@ function block_core_navigation_render_submenu_icon() {
 
 
 /**
- * Finds the first non-empty `wp_navigation` Post.
+ * Finds the most recent published `wp_navigation` Post.
  *
  * @return WP_Post|null the first non-empty Navigation or null.
  */
-function block_core_navigation_get_first_non_empty_navigation() {
-	// Order and orderby args set to mirror those in `wp_get_nav_menus`
-	// see:
-	// - https://github.com/WordPress/wordpress-develop/blob/ba943e113d3b31b121f77a2d30aebe14b047c69d/src/wp-includes/nav-menu.php#L613-L619.
-	// - https://developer.wordpress.org/reference/classes/wp_query/#order-orderby-parameters.
+function block_core_navigation_get_most_recent_published_navigation() {
+	// We default to the most recently created
 	$parsed_args = array(
 		'post_type'      => 'wp_navigation',
 		'no_found_rows'  => true,
-		'order'          => 'ASC',
-		'orderby'        => 'name',
+		'order'          => 'DESC',
+		'orderby'        => 'date',
 		'post_status'    => 'publish',
-		'posts_per_page' => 20, // Try the first 20 posts.
+		'posts_per_page' => 1, // Try the first 20 posts.
 	);
 
-	$navigation_posts = new WP_Query( $parsed_args );
-	foreach ( $navigation_posts->posts as $navigation_post ) {
-		if ( has_blocks( $navigation_post ) ) {
-			return $navigation_post;
-		}
+	$navigation_post = new WP_Query( $parsed_args );
+	if( count( $navigation_post->posts )  > 0 ) {
+		return $navigation_post->posts[0];
```

Comment:
```suggestion
		return $navigation_post->post;
```

This should work

Reply:
Why would it? WP_Query returns an array with one element.

Reply:
https://github.com/WordPress/wordpress-develop/blob/67023d0c66267420107bb565f887ae6a95b12666/src/wp-includes/class-wp-query.php#L118-L127

But yes maybe that's not intended for this use case.

Reply:
I feel like `post` was for `the_post()` and I think this clunkiness `posts[0]` makes it clear we're only about one post.